pkg/collector/corechecks/oracle/statements_test.go

```diff
@@ -126,3 +126,49 @@ func TestUInt64Binding(t *testing.T) {
 		assert.Equalf(t, retValue, 1, "IN uint64")
 	}
 }
+
+func TestLongMultibyteQuery(t *testing.T) {
+	c, _ := newDefaultCheck(t, "", "")
+	defer c.Teardown()
+
+	c.dbmEnabled = true
+
+	// This is to ensure that query samples return rows
+	c.config.QuerySamples.IncludeAllSessions = true
+
+	c.statementsLastRun = time.Now().Add(-48 * time.Hour)
+	err := c.Run()
+	assert.NoError(t, err, "check run")
+
+	// We're just testing the agent's ability to handle a SQL query whose bytes exceed 4k
+	// when there's less than 4k characters. Actual result doesn't matter so long as it eventually collects
+	// the query
+	longQuery := "SELECT sid AS "
+	for i := 0; i < 1500; i++ {
+		longQuery += "氏"
+	}
+	longQuery += " FROM v$session LIMIT 1;"
+
+	_, err = c.db.Exec(longQuery)
```

Comment:
`Exec` will consume all the rows, and therefore disassociate the query with the session, so the query won't be captured in samples. Instead, we can execute the query without scanning any rows.